lib/msal-browser/src/naa/mapping/NestedAppAuthAdapter.ts

```diff
@@ -0,0 +1,198 @@
+         */
+        const tokenRequest: TokenRequest = {
+            userObjectId: request.account?.homeAccountId,
+            clientId: this.clientId,
+            authority: request.authority,
+            scope: request.scopes.join(" "),
+            correlationId:
+                request.correlationId !== undefined
+                    ? request.correlationId
+                    : "",
+            prompt: request.prompt !== undefined ? request.prompt : "",
+            nonce: request.nonce !== undefined ? request.nonce : "",
+            claims: request.claims !== undefined ? request.claims : "",
+            state: request.state !== undefined ? request.state : "",
+            authenticationScheme:
+                request.authenticationScheme !== undefined
+                    ? request.authenticationScheme
+                    : "",
+            clientCapabilities: undefined,
+            extraParameters: extraParams,
+        };
+
+        return tokenRequest;
+    }
+
+    public fromNaaTokenResponse(
+        request: TokenRequest,
+        response: TokenResponse
+    ): AuthenticationResult {
+        const expiresOn = new Date(
+            TimeUtils.nowSeconds() + (response.expires_in || 0) * 1000
+        );
+        const authToken: AuthToken = new AuthToken(
+            response.id_token,
+            this.crypto
+        );
+        const authenticationResult: AuthenticationResult = {
+            authority: response.account.environment,
+            uniqueId: response.account.homeAccountId,
+            tenantId: response.account.tenantId,
+            scopes: response.scope.split(" "),
+            account: this.fromNaaAccountInfo(response.account),
+            idToken: response.id_token,
+            idTokenClaims: authToken.claims,
+            accessToken: response.access_token,
+            fromCache: true,
+            expiresOn: expiresOn,
+            tokenType:
+                request.authenticationScheme !== undefined
+                    ? request.authenticationScheme
+                    : "Bearer",
+            correlationId: request.correlationId,
+            requestId: "",
+            extExpiresOn: expiresOn,
+            state: response.state,
+        };
+
+        return authenticationResult;
+    }
+
+    public fromNaaAccountInfo(fromAccount: NaaAccountInfo): MsalAccountInfo {
+        const account: MsalAccountInfo = {
+            homeAccountId: fromAccount.homeAccountId,
+            environment: fromAccount.environment,
+            tenantId: fromAccount.tenantId,
+            username: fromAccount.username,
+            localAccountId: fromAccount.localAccountId,
+            name: fromAccount.name,
+            idTokenClaims: {},
+        };
+
+        return account;
+    }
+
+    /**
+     *
+     * @param error BridgeError
+     * @returns AuthError, ClientAuthError, ClientConfigurationError, ServerError, InteractionRequiredError
+     */
+    public fromBridgeError(
```

Comment:
TODO - Need to review these error transformations with somebody from the MSAL.js team to see if they can live this